plaso/engine/artifact_filters.py

```diff
@@ -0,0 +1,182 @@
+    self._artifacts_registry = artifacts_registry
+    self._artifacts = artifacts
+    self._knowledge_base = knowledge_base
+
+  def BuildFindSpecs(self, environment_variables=None):
+    """Builds find specification from a forensic artifact definitions.
+
+    Args:
+      environment_variables (Optional[list[EnvironmentVariableArtifact]]):
+          environment variables.
+    """
+    find_specs = {}
+
+    artifact_definitions = []
+    for artifact_filter in self._artifacts:
+      if self._artifacts_registry.GetDefinitionByName(artifact_filter):
+        artifact_definitions.append(
+            self._artifacts_registry.GetDefinitionByName(artifact_filter))
+
+    for definition in artifact_definitions:
+      for source in definition.sources:
+        if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+          for path_entry in source.paths:
+            self.BuildFindSpecsFromFileArtifact(
+                path_entry, source.separator, environment_variables,
+                self._knowledge_base.user_accounts, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+          keys = set(source.keys)
+          for key_entry in keys:
+            if self._CheckKeyCompatibility(key_entry):
+              self.BuildFindSpecsFromRegistryArtifact(
+                  key_entry, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+          # TODO: Handle Registry Values Once Supported in dfwinreg.
+          # https://github.com/log2timeline/dfwinreg/issues/98
+          logging.warning(('Unable to handle Registry Value, extracting '
+                           'key only: "{0:s}"').format(source.key_value_pairs))
+
+          for key_pair in source.key_value_pairs:
+            keys = set()
+            keys.add(key_pair.get('key'))
+            for key_entry in keys:
+              if self._CheckKeyCompatibility(key_entry):
+                self.BuildFindSpecsFromRegistryArtifact(key_entry, find_specs)
+        else:
+          logging.warning(('Unable to handle artifact, plaso does not '
+                           'support: "{0:s}"').format(source.type_indicator))
+
+    self._knowledge_base.SetValue(self.ARTIFACT_FILTERS, find_specs)
+
+  @classmethod
+  def BuildFindSpecsFromFileArtifact(
+      cls, path_entry, separator, environment_variables, user_accounts,
+      find_specs):
+    """Builds find specifications from a file artifact definition.
+
+    Args:
+      path_entry (str): current file system path to add.
+      separator (str): file system path segment separator.
+      environment_variables list(str): environment variable attributes used to
+          dynamically populate environment variables in key.
+      user_accounts (list[str]): identified user accounts stored in the
+          knowledge base.
+      find_specs (dict[artifacts.artifact_types]):  Dictionary containing
```

Comment:
`find_specs (dict[artifacts.artifact_types]):  Dictionary containing ...` I had a brief look at this but this arg looks weird to me. Find specs are a dfVFS class, but type information says artifacts.artifact_types ?